crates/transaction-pool/src/blobstore/disk.rs

```diff
@@ -194,9 +193,56 @@ impl BlobStore for DiskFileBlobStore {
 
     fn get_by_versioned_hashes_v2(
         &self,
-        _versioned_hashes: &[B256],
+        versioned_hashes: &[B256],
     ) -> Result<Option<Vec<BlobAndProofV2>>, BlobStoreError> {
-        Ok(None)
+        let mut result = vec![None; versioned_hashes.len()];
+
+        for (_tx_hash, blob_sidecar) in self.inner.blob_cache.lock().iter() {
+            if let Some(blob_sidecar) = blob_sidecar.as_eip7594() {
+                for (hash_idx, match_result) in
+                    blob_sidecar.match_versioned_hashes(versioned_hashes)
+                {
+                    result[hash_idx] = Some(match_result);
```

Comment:
I think this entire fn is more or less the same code as v1, only this here is different, so I believe we can unify if try to offload this to a function that accepts a closure for this section

Reply:
pushed do check it out!